crates/uv-client/src/registry_client.rs

```diff
@@ -337,12 +332,26 @@ impl RegistryClient {
                     let _permit = download_concurrency.acquire().await;
                     match index.format {
                         IndexFormat::Simple => {
-                            if let Some(metadata) = self
-                                .simple_single_index(package_name, index.url, capabilities)
+                            let status_code_strategy =
+                                self.index_urls.status_code_strategy_for(index.url);
+                            match self
+                                .simple_single_index(
+                                    package_name,
+                                    index.url,
+                                    capabilities,
+                                    &status_code_strategy,
+                                )
                                 .await?
                             {
-                                results.push((index.url, MetadataFormat::Simple(metadata)));
-                                break;
+                                SimpleMetadataSearchOutcome::Found(metadata) => {
+                                    results.push((index.url, MetadataFormat::Simple(metadata)));
+                                    break;
+                                }
+                                // Package not found, so we will continue on to the next index (if there is one)
+                                SimpleMetadataSearchOutcome::NotFound => {}
+                                // The search failed because of an HTTP status code that we don't ignore for
+                                // this index. We end our search here.
+                                SimpleMetadataSearchOutcome::StatusCodeFailure => break,
```

Comment:
Should this variant include the status code? Should we log this interaction here?

Reply:
Having trouble following a bit, how does this error propagate upwards?

Reply:
> Having trouble following a bit, how does this error propagate upwards?

It's the same as before. On 401 or 403, we set that in the `IndexCapabilities`. When you break here, that effectively ends the search and a `PackageNotFound` error will be returned at the end of this function.